paasta_tools/cli/cmds/local_run.py

```diff
@@ -861,7 +861,21 @@ def configure_and_run_docker_container(
     )
 
 
+def reexecute_pull_with_sudo():
+    paasta_print("Re-executing paasta local-run --pull with sudo..")
```

Comment:
This is good, and does the same thing as the wrapper.

However, can you do even better by adding in the `sudo -H` *exactly* where we need it, which is wherever we shell out to an actual docker command.

Reply:
As we discussed I added "-H", made it simpler and fixed tests.